### ISSUE TITLE
Review of the lab10 solution - done by Francesca Zafonte

### ISSUE THREAD

**Issue by Zafonte** [baraccothomas/computational-intelligence#8]
Hi Thomas, I reviewed your code and I'm glad to share my feedback.

_**CODE:**_
- The code is well written and commented.

- I suggest you rename your agent to QPlayer so that it is clearer what you are implementing for him.
  About the Q-learning algorithm, I like your implementation and I found it very clarify.  Congratulations on your excellent 
  achievement.

- According to the q-table update formula of the q-learning algorithm you should use next_state. 
  Below is the correct formula:
  ```python
  self.Q[tuple(state.get_board())][action] += self.alpha * (reward + self.gamma * np.max(self.Q[tuple(next_state.get_board())]) - 
  self.Q[tuple(state.get_board())][action])
  ```

To summarize, I think that the problem was carried out very well. 
I hope that you'll find my little tips useful! Good job!
